### PR TITLE
Adding initial f32 (and placeholder f64) support to the VM.

### LINKED ISSUE #5546: Implement the F32/F64 VM instruction extensions.
Though there's a carve-out for them the instructions aren't implemented yet:
https://github.com/google/iree/blob/36db60039462a097d04cab99cbe91d424fef3ea8/iree/vm/bytecode_dispatch.c#L1464-L1468

A dozen or so arithmetic ops is likely all we need to start, as well as the type conversion ala `ExtI32I64S` for int<->float.

These will be needed for VMVX but also be useful in on the host for when we are detensoring and may end up with floating point operations there.

### PR DESCRIPTION
**Pull request by benvanik**
This is largely a placeholder for the full set of floating-point ops we'll want to support but defines enough of a skeleton that'll be easy to flesh out. This enables VMVX and is required when we are detensoring and have float value manipulation on the host.

Lots of boilerplate here. Would be nice to refactor things to prevent this explosion now that we've got the full type set specified. There's some template goo that could probably be used but the APInt/APFloat type system is unwieldy for this use.

Fixes #5546.

### CODE REVIEW

File: iree/compiler/Dialect/VM/IR/VMOps.td [hunk 1]
```diff
@@ -1197,19 +1204,58 @@ class VM_SwitchPrimitiveOp<I type, string mnemonic, VM_OPC opcode,
...
+      AllTypesMatch<["default_value", "result"]>,
+    ])> {
+  let description = [{
+    Returns the value with the given `index` in `values` or `default_value` if
+    the index is out of bounds.
+
+    ```mlir
+    // Switch %arg0 to cases of %c100/%c200/%c300 if arg0==0, ==1, ==2.
+    // If %arg0 is out of range (<0 or >2) then default to %c5.
+    %0 = vm.switch.f32 %index[%c100, %c200, %c300] else %c5 : f32
```
Comment (ScottTodd) [inline iree/compiler/Dialect/VM/IR/VMOps.td:1369]: `%arg0` or `%index`?

**Review by ScottTodd — COMMENTED** [review]
reviewed the first two commits, will continue the review later

File: iree/vm/stack.c [hunk 2]
```diff
@@ -419,14 +419,14 @@ IREE_API_EXPORT iree_status_t IREE_API_CALL iree_vm_stack_function_enter(
...
-  IREE_TRACE_ZONE_APPEND_VALUE(z0, frame_size);
-#endif  // IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION
+  // #if IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION
+  //   // TODO(benvanik): cache source location and query from module.
+  //   iree_string_view_t function_name = iree_vm_function_name(function);
+  //   IREE_TRACE_ZONE_BEGIN_NAMED_DYNAMIC(z0, function_name.data,
+  //                                       function_name.size);
+  //   callee_frame->trace_zone = z0;
+  //   IREE_TRACE_ZONE_APPEND_VALUE(z0, frame_size);
+  // #endif  // IREE_TRACING_FEATURES & IREE_TRACING_FEATURE_INSTRUMENTATION
```
Comment (ScottTodd) [inline iree/vm/stack.c:429]: Why's this commented out?

**Review by ScottTodd — APPROVED** [review]

File: iree/vm/stack.c (hunk 2 above)
Comment (benvanik) [reply iree/vm/stack.c:429]: good catch!

**Pull request merged**